src/lyra.ts

```diff
@@ -648,15 +649,22 @@ export function search<S extends PropertiesSchema>(
     for (let j = 0; j < tokensLength; j++) {
       const term = tokens[j];
       const documentIDs = getDocumentIDsFromSearch(lyra, { ...params, index, term, exact });
-      const termOccurrencies = lyraOccurrencies[term];
+      
+      // lyraOccurrencies[term] can be undefined, 0, string, or { [k: string]: number }
```

Comment:
- `termOccurrencies`'s type is inferred as `number`, but it's actually `number | string | { [k: string]: number } | undefined`

I recommend the maintainers to verify why this inferred type is misleading after this PR gets merged.